### DISCUSSION TITLE
Tailwind v1.7 is now available! Gradients, background-clip, improved `@apply`, and more...

### DISCUSSION THREAD

**Discussion by adamwathan** [tailwindlabs/tailwindcss#2183] [Show and tell]
Tailwind CSS v1.7 is now available with gradients, background-clip support, a new experimental version of `@apply` that works with any class (!), and tons more:

### New features

- [Gradients](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#gradients)
- [New background-clip utilities](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#new-background-clip-utilities)
- [New gap utility aliases](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#new-gap-utility-aliases)
- [New `contents` display utility](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#new-contents-display-utility)
- [Default letter-spacing per font-size](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#default-letter-spacing-per-font-size)
- [Divide border styles](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#divide-border-styles)
- [Access entire config object from plugins](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#access-entire-config-object-from-plugins)
- [Define colors as closures](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#define-colors-as-closures)

### Deprecations

- [Deprecated gap utilities](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#deprecated-gap-utilities)

### Experimental features

- [Use `@apply` with variants and other complex classes](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#use-apply-with-variants-and-other-complex-classes)
- [New color palette](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#new-color-palette)
- [Extended spacing scale](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#extended-spacing-scale)
- [Default line-heights per font-size by default](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#default-line-heights-per-font-size-by-default)
- [Extended font size scale](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#extended-font-size-scale)

For all the details, [check out the complete release notes on GitHub](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0).

**Comment by arskhatri5** [comment]
Hello respected Sir,

can you please add some padding classes on hover.
e.g  hover:pl-4

and any link on hover  dropdown class display block..
i'm create a large project using only tailwind css after creating i'm send you link and i think you feel proud.
 
Thank You.
